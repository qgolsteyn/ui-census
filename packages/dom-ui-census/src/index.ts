import createDomCensusObjectFactory from "./createDomCensusObjectFactory";

import baseAdapter from "./definitions/baseAdapter";
import htmlAdapter from "./definitions/htmlAdapter";

export { createDomCensusObjectFactory, baseAdapter, htmlAdapter };
