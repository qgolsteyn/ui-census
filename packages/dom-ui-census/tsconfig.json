{
  "extends": "../../tsconfig-base",
  "compilerOptions": {
    "outDir": "./lib/",
    "rootDir": "./src",
    "tsBuildInfoFile": ".tsc/.tsbuildinfo"
  },
  "references": [{ "path": "../ui-census" }],
  "include": ["src"]
}
