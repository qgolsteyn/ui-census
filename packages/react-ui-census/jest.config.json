{
  "moduleFileExtensions": ["ts", "tsx", "js"],
  "transform": {
    "\\.tsx?$": "ts-jest"
  },
  "testMatch": ["**/*.test.tsx", "**/*.test.ts"],
  "globals": {
    "ts-jest": {
      "tsConfig": "tsconfig.json"
    }
  },
  "moduleNameMapper": {
    "dom-ui-census/(.*)$": "<rootDir>/../dom-ui-census/src/$1",
    "dom-ui-census$": "<rootDir>/../dom-ui-census/src",
    "ui-census/(.*)$": "<rootDir>/../ui-census/src/$1",
    "ui-census$": "<rootDir>/../ui-census/src"
  }
}
