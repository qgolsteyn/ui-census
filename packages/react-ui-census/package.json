{
  "name": "react-ui-census",
  "version": "0.0.11",
  "description": "UI Census is a Javascript library for testing user interfaces",
  "keywords": [
    "UI",
    "testing"
  ],
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "author": {
    "name": "Quentin Golsteyn",
    "email": "qgolsteyn@gmail.com",
    "url": "https://golsteyn.com"
  },
  "scripts": {
    "build": "tsc --build ./tsconfig.json",
    "test": "jest --config=./jest.config.json"
  },
  "license": "MIT",
  "files": [
    "lib"
  ],
  "repository": {
    "url": "https://github.com/qgolsteyn/ui-census"
  },
  "dependencies": {
    "ui-census": "^0.0.11"
  },
  "peerDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1"
  },
  "devDependencies": {
    "react": "^16.13.1",
    "react-dom": "^16.13.1"
  }
}
