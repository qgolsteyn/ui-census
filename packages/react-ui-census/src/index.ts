import createReactAdapter from "./createReactAdapter";
import createTestRender from "./createTestRender";

export { CensusObject, CensusDefinition } from "ui-census";

export { createReactAdapter, createTestRender };
