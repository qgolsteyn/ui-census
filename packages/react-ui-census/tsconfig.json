{
  "extends": "../../tsconfig-base",
  "compilerOptions": {
    "outDir": "./lib/",
    "rootDir": "./src",
    "baseUrl": ".",
    "tsBuildInfoFile": ".tsc/.tsbuildinfo"
  },
  "references": [{ "path": "../ui-census" }],
  "include": ["src"],
  "exclude": ["**/__test__"]
}
