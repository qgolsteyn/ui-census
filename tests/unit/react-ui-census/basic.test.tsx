import { createDOMAdapter } from "dom-ui-census";
import React from "react";
import { createReactAdapter } from "react-ui-census";

const adapter = createDOMAdapter(
  (target) => {
    let elements = Array.from(target.querySelectorAll("div"));

    return elements;
  },
  {
    text: (element) => element.textContent,
  },
  {}
);

const divReactAdapter = createReactAdapter(adapter);

let container: HTMLDivElement | null;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container!);
});

afterEach(() => {
  document.body.removeChild(container!);
  container = null;
});

test("it handles a basic query", () => {
  const divAccessor = divReactAdapter(<div>Test 1</div>, container!);

  expect(divAccessor().single().text).toBe("Test 1");
});

test("it handles a query by prop with mapping", () => {
  const divAccessor = divReactAdapter(
    <>
      <div>1</div>
      <div>2</div>
      <div>3</div>
    </>,
    container!
  );

  expect(
    divAccessor()
      .all()
      .map((node) => node.text)
  ).toStrictEqual(["1", "2", "3"]);
});

test("it supports snapshot testing", () => {
  const divAccessor = divReactAdapter(<div id="test">Test 4</div>, container!);

  expect(divAccessor().all()).toMatchSnapshot();
});

test("it supports snapshot testing with multiple elements", () => {
  const divAccessor = divReactAdapter(
    <>
      <div id="test">Test 1</div>
      <div id="test">Test 2</div>
      <div id="test">
        Test 3 and <div id="test">Test 4</div>
      </div>
    </>,
    container!
  );

  expect(divAccessor().all()).toMatchSnapshot();
});
