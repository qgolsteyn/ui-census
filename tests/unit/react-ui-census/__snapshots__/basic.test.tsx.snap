// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it supports snapshot testing 1`] = `
Array [
  Object {
    "text": "Test 4",
  },
]
`;

exports[`it supports snapshot testing with multiple elements 1`] = `
Array [
  Object {
    "text": "Test 1",
  },
  Object {
    "text": "Test 2",
  },
  Object {
    "text": "Test 3 and Test 4",
  },
  Object {
    "text": "Test 4",
  },
]
`;
